### PR TITLE
🧑‍⚕️Handle transaction errors

### LINKED ISSUE #33: Token transfer: handle transaction error
Handle transaction errors:

- read [transfer events](https://github.com/orgs/Joystream/projects/38) to see if the transaction was successful
- decide if the wait screen should be closed on "finalized" or "inBlock" state
- see potential workflow improvements (deduplicate code for similar action)

### LINKED ISSUE #190: Buy membership flow
@[figma](https://www.figma.com/file/J2KUqGu97lNQC2iKsxJeVH/My-Profile?node-id=192%3A13077)

Finish existing "create member" modal flow for "buy membership" case. The modal, changing modal states and valiations were already implemented.

### Requirements

- only "not reffered" flow
- handle transaction error & success.

### PR DESCRIPTION
**Pull request by jodator**
Closes #33. Closes #190.

*   I've added error & success checks based on the [events from the blockchain](https://polkadot.js.org/docs/api/examples/promise/transfer-events).
*   The odd thing that bugged me was that extension was not loading the accounts every time (see this PR: #210)

### CODE REVIEW

File: packages/ui/test/modals/AddMembershipModal.test.tsx [hunk 1]
```diff
@@ -144,27 +145,51 @@ describe('UI: AddMembershipModal', () => {
...
+      it('Renders transaction success', async () => {
+        const { getByText, findByText } = await renderAuthorizeStep()
+        fireEvent.click(getByText(/^sign and create a member$/i))
+
+        expect(await findByText('Success')).toBeDefined()
+        expect(getByText(/^realbobbybob/i)).toBeDefined()
+      })
+    })
+
+    describe('Failure', () => {
```
Comment (krzysztof-jelski) [inline packages/ui/test/modals/AddMembershipModal.test.tsx:172]: This seems like a duplicate of the previous test and does not describe failure but success.

File: packages/ui/src/modals/AddMembershipModal/AddMembershipFailureModal.tsx [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
-}
+export const AddMembershipFailureModal = ({ onClose, member }: Props) => (
+  <Modal modalSize="xs" modalHeight="s">
+    <ResultModalBody>
+      <CloseSmallModalButton onClick={onClose} />
+      <FailureIcon />
+      <ModalTitle>
+        <span className="red-title">Oh no!</span> Failure
+      </ModalTitle>
+      <ResultText size={2}>There was a problem with crating a membership for {member.name}.</ResultText>
```
Comment (krzysztof-jelski) [inline packages/ui/src/modals/AddMembershipModal/AddMembershipFailureModal.tsx:19]:
```suggestion
      <ResultText size={2}>There was a problem with creating a membership for {member.name}.</ResultText>
```

File: packages/ui/src/components/page/Sidebar/CurrentMember/CurrentMember.tsx [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 import { AddMembershipButton } from '../../../AddMembershipButton'
+import { ArrowDownExpandedIcon, ArrowIcon } from '../../../icons'
 import { MemberDarkHover, MemberInfo } from '../../../MemberInfo'
+import { MembershipsCount } from '../../../MembershipCount'
 import { SwitchMemberModal } from './SwitchMemberModal'
 
 export const CurrentMember = () => {
   const { count, members, active } = useMembership()
   const [isOpen, setIsOpen] = useState(false)
+  const [isCreateOpen, setIsCreateOpen] = useState(false)
```
Comment (krzysztof-jelski) [inline packages/ui/src/components/page/Sidebar/CurrentMember/CurrentMember.tsx:15]: As it's a very common pattern, it's worth extracting a hook:
`const [isOpen, toggleOpen] = useToggle()`

File: packages/ui/src/modals/AddMembershipModal/AddMembershipModal.tsx [hunk 4]
```diff
@@ -14,47 +15,63 @@ interface MembershipModalProps {
...
+  }, [subscription])
 
   const onSubmit = (params: Member) => {
-    setState('Authorize')
+    setStep('AUTHORIZE')
     setParams(params)
   }
 
   const onSign = (transaction: Observable<ISubmittableResult>) => {
     const statusCallback = (result: ISubmittableResult) => {
```
Comment (krzysztof-jelski) [inline packages/ui/src/modals/AddMembershipModal/AddMembershipModal.tsx:40]: The `statusCallback` and `ModalState` are now duplicates of the ones in `TransferModal`. Worth extracting.

**Review by krzysztof-jelski — CHANGES_REQUESTED** [review]
As per single comments

File: packages/ui/test/modals/AddMembershipModal.test.tsx (hunk 1 above)
Comment (jodator) [reply packages/ui/test/modals/AddMembershipModal.test.tsx:172]: Oh, my bad :( I'll fix that. Thanks.

**Review by krzysztof-jelski — APPROVED** [review]

**Pull request merged**